tools/xdp/s2n-quic-xdp/src/if_xdp.rs

```diff
@@ -0,0 +1,265 @@
+// Copyright Intel Corporation
+// SPDX-License-Identifier: GPL-2.0 WITH Linux-syscall-note
+// Modifications copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::Result;
+use bitflags::bitflags;
+use core::{ffi::CStr, mem::size_of};
+
+bitflags!(
+    /// Options for the `flags` field in [`Address`]
+    ///
+    /// See https://github.com/torvalds/linux/blob/2bac7dc169af3cd4a0cb5200aa1f7b89affa042a/include/uapi/linux/if_xdp.h#L16-L27
+    #[derive(Copy, Clone, Debug, Default)]
+    #[repr(transparent)]
+    pub struct XdpFlags: u16 {
+        const SHARED_UMEM = 1 << 0;
+        /// Force copy mode
+        const COPY = 1 << 1;
+        /// Force zero-copy mode
+        const ZEROCOPY = 1 << 2;
+        /// If this option is set, the driver might go to sleep and in that case the
+        /// XDP_RING_NEED_WAKEUP flag in the fill and/or Tx rings will be set.
+        ///
+        /// If it is set, the application needs to explicitly wake up the driver with a `poll()` for
+        /// Rx or `sendto()` for Tx. If you are running the driver and the application on the same
+        /// core, you should use this option so that the kernel will yield to the user space
+        /// application.
+        const USE_NEED_WAKEUP = 1 << 3;
+    }
+);
+
+bitflags!(
+    /// Flags for the umem config
+    ///
+    /// See https://github.com/torvalds/linux/blob/2bac7dc169af3cd4a0cb5200aa1f7b89affa042a/include/uapi/linux/if_xdp.h#L30
+    #[derive(Copy, Clone, Debug, Default)]
+    #[repr(transparent)]
+    pub struct UmemFlags: u32 {
+        const UNALIGNED_CHUNK_FLAG = 1 << 0;
+    }
```

Comment:
All of the flags are `u32`. You can also see the [config field in libxdp](https://github.com/xdp-project/xdp-tools/blob/a76e7a2b156b8cfe38992206abe9df1df0a29e38/headers/xdp/xsk.h#L203). I can link this if that's helpful.

Reply:
So if I understand correctly, this is specifically matching the xdp-project interface. Yea it would be good to include a link to the xdp-project github, since this is definitely new territory for me and possible others.